operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerEventSource.java

```diff
@@ -154,19 +158,28 @@ public void onDelete(R resource, boolean b) {
   private synchronized void onAddOrUpdate(Operation operation, R newObject, R oldObject,
       Runnable superOnOp) {
     var resourceID = ResourceID.fromResource(newObject);
-    if (eventRecorder.isRecordingFor(resourceID)) {
-      log.debug("Recording event for: {}", resourceID);
-      eventRecorder.recordEvent(newObject);
-      return;
+
+    String previous = newObject.getMetadata().getAnnotations()
+        .get(KubernetesDependentResource.PREVIOUS_ANNOTATION_KEY);
+    boolean known = false;
+    if (previous != null) {
+      String[] parts = previous.split(",");
+      if (id.equals(parts[0])) {
+        if (oldObject == null && parts.length == 1) {
+          known = true;
+        } else if (oldObject != null && parts.length == 2
+            && oldObject.getMetadata().getResourceVersion().equals(parts[1])) {
+          known = true;
+        }
+      }
     }
-    if (temporaryCacheHasResourceWithSameVersionAs(newObject)) {
+    superOnOp.run();
```

Comment:
This is little suspicious, is't this call removes the resource from the cache, but we then compare that in the next step?

Reply:
You're right, before committing I just scanned the code again and saw that was in each branch and moved it up.  I'll put it back.